src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/TestStatusTest.java

```diff
@@ -19,20 +23,31 @@ void when_edifactStringDoesNotStartWithCorrectKey_expect_illegalArgumentExceptio
 
     @Test
     void when_edifactStringIsPassed_expect_returnATestStatusObject() {
-        assertThat(testStatus)
-                .usingRecursiveComparison()
-                .isEqualTo(TestStatus.fromString("STS++CO"));
-    }
+        var valueMap = Map.of(
+            TestStatusCode.CORRECTED, "CO",
+            TestStatusCode.UNKNOWN, "UN",
+            TestStatusCode.AMENDED, "AM",
+            TestStatusCode.CANCELLED, "CA",
+            TestStatusCode.ENTERED_IN_ERROR, "EN",
+            TestStatusCode.FINAL, "FI",
+            TestStatusCode.PRELIMINARY, "PR",
+            TestStatusCode.REGISTERED, "RE");
 
-    @Test
-    void when_mappingSegmentObjectToEdifactString_expect_returnCorrectEdifactString() {
-        String expectedEdifactString = "STS+CO'";
+        assertThat(valueMap).hasSameSizeAs(TestStatusCode.values());
 
-        TestStatus testStatus = TestStatus.builder()
-                .testStatusCode(TestStatusCode.CORRECTED)
-                .build();
+        for (TestStatusCode testStatusCode : TestStatusCode.values()) {
+            assertThat(new TestStatus(testStatusCode)).usingRecursiveComparison()
+                .isEqualTo(TestStatus.fromString("STS++" + valueMap.get(testStatusCode)));
+        }
```

Comment:
This loos like we want a `@ParameterizedTest` with an `@EnumSource` or `@CsvSource`

Reply:
Can't use EnumSource as I'm also telling what's the expected short code. I don't want to use the value from enum to not do a self-check. Let's leave as is ok?

Reply:
Sure